applications/wobbly/templates/cronjob-maintenance.yaml

```diff
@@ -0,0 +1,164 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: "wobbly-maintenance"
+  labels:
+    {{- include "wobbly.labels" . | nindent 4 }}
+spec:
+  schedule: {{ .Values.maintenance.schedule | quote }}
+  concurrencyPolicy: "Forbid"
+  jobTemplate:
+    spec:
+      activeDeadlineSeconds: {{ .Values.maintenance.deadlineSeconds }}
+      ttlSecondsAfterFinished: {{ .Values.maintenance.cleanupSeconds }}
+      template:
+        metadata:
+          {{- with .Values.podAnnotations }}
+          annotations:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          labels:
+            {{- include "wobbly.selectorLabels" . | nindent 12 }}
+            app.kubernetes.io/component: "maintenance"
+        spec:
+          {{- with .Values.affinity }}
+          affinity:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          {{- if .Values.cloudsql.enabled }}
+          serviceAccountName: "wobbly"
+          {{- else }}
+          automountServiceAccountToken: false
+          {{- end }}
+          containers:
+            {{- if .Values.cloudsql.enabled }}
+            - name: "cloud-sql-proxy"
+              # Running the sidecar as normal causes it to keep running and
+              # thus the Pod never exits, the Job never finishes, and the hook
+              # blocks the sync. Have the main pod signal the sidecar by
+              # writing to a file on a shared emptyDir file system, and use a
+              # simple watcher loop in shell in the sidecar container to
+              # terminate the proxy when the main container finishes.
+              #
+              # Based on https://stackoverflow.com/questions/41679364/
```

Comment:
Could we use a [sidecar container :tm: ](https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/#jobs-with-sidecar-containers) for this? I think they're available  in k8s >= 1.29, which .....

Looks like usdf prod at least is 1.27 :cry: so nevermind.

Reply:
Ooo, I didn't know about those! Those will simplify a lot of patterns with Cloud SQL. As soon as we have 1.29 everywhere, I will definitely switch.